core/local/steps/recurse.js

```diff
@@ -0,0 +1,25 @@
+/* @flow */
+
+/*::
+import type Buffer from './buffer'
+import type LinuxProducer from './linux_producer'
+
+export interface Adder {
+  add(string): Promise<*>,
+}
+*/
+
+module.exports = function (buffer /*: Buffer */, opts /*: { adder: Adder } */) /*: Buffer */ {
+  return buffer.asyncMap(async (batch) => {
+    for (const event of batch) {
+      if (event.docType === 'directory' && event.action === 'created') {
```

Comment:
Last one: *recurse* feels a bit abstract, since we're only doing something recursively with subdirs... *handleSubdirs* or *watchSubdirs* maybe?

Reply:
(the previous source/producer's `#watch()` method name was clearer than `#add()` to me)